### ISSUE TITLE
Flutter/VSCode/Dart : unused_field and unused_element errors for private fields in classes

### ISSUE THREAD

**Issue by vin-fandemand** [dart-lang/linter#2184]
**Describe the issue**
Consider the following snippet
```
class TestClass {
  final String field;
  String _field;

  TestClass(this.field);

  void method() {
    _field = '';
  }
}
```
The private field `_field` is not detected as being used although it is being used inside one of the methods. I have seen it quite often when I develop in Flutter where I often initialize a private field inside an `initState` method for a `StatefulWidget`. Is this a bug or am I not understanding it correctly?
![image](https://user-images.githubusercontent.com/64597233/87196444-0157cb00-c2c8-11ea-875d-633be2868168.png)

**To Reproduce**
```
class TestClass {
  final String field;
  String _field;

  TestClass(this.field);

  void method() {
    _field = '';
  }
}
```

**Expected behavior**
The private field is used in the methods. So ideally shouldn't show as unused.

**Additional context**
https://github.com/Dart-Code/Dart-Code/issues/2630
